modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1536,6 +1583,355 @@ MULC_SIMD(float, float)
+                                        v_round(v_select(v_mask2, v_zero, div_op(s_tag, vg_load_f32(&in[x + 7*nlanes/4]), s2, v_scale))))));
+
+            vx_store(&out[x + 2 * nlanes],
+                        v_pack_u(v_pack(v_round(v_select(v_mask3, v_zero, div_op(s_tag, vg_load_f32(&in[x + 2*nlanes]), s3, v_scale))),
+                                        v_round(v_select(v_mask1, v_zero, div_op(s_tag, vg_load_f32(&in[x + 9*nlanes/4]), s1, v_scale)))),
+                                 v_pack(v_round(v_select(v_mask2, v_zero, div_op(s_tag, vg_load_f32(&in[x + 5*nlanes/2]), s2, v_scale))),
+                                        v_round(v_select(v_mask3, v_zero, div_op(s_tag, vg_load_f32(&in[x + 11*nlanes/4]), s3, v_scale))))));
+        }
+
+        if (x < length)
+        {
+            x = length - lanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC>
+CV_ALWAYS_INLINE int divc_simd_c3_impl(scale_tag_t s_tag, const SRC* in, float* out,
+                                       const v_float32& s1, const v_float32& s2,
+                                       const v_float32& s3, const v_float32& v_scale, const int length,
+                                       const int nlanes, const int lanes)
+{
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - lanes; x += lanes)
+        {
+            v_float32 a1 = vg_load_f32(&in[x]);
+            v_float32 a2 = vg_load_f32(&in[x + nlanes]);
+            v_float32 a3 = vg_load_f32(&in[x + 2*nlanes]);
+
+            vx_store(&out[x], div_op(s_tag, a1, s1, v_scale));
+            vx_store(&out[x + nlanes], div_op(s_tag, a2, s2, v_scale));
+            vx_store(&out[x + 2*nlanes], div_op(s_tag, a3, s3, v_scale));
+        }
+
+        if (x < length)
+        {
+            x = length - lanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC, typename DST>
+CV_ALWAYS_INLINE int divc_mask_simd_c3(scale_tag_t s_tag, const SRC in[],
+                                       const float scalar[], DST out[],
+                                       const int length, const float scale)
+{
+    constexpr int chan = 3;
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+    constexpr int lanes = chan * nlanes;
+
+    if (length < lanes)
+        return 0;
+
+    v_float32 v_scale = vx_setall_f32(scale);
+
+    v_float32 s1 = vx_load(scalar);
+#if CV_SIMD_WIDTH == 32
+    v_float32 s2 = vx_load(&scalar[2]);
+    v_float32 s3 = vx_load(&scalar[1]);
+#else
+    v_float32 s2 = vx_load(&scalar[1]);
+    v_float32 s3 = vx_load(&scalar[2]);
```

Comment:
by why reorderning is here? please add comment

Reply:
> by why reorderning is here? please add comment

The difference links with lengths of SIMD vectors of SSE/NEON(128 bits), AVX2(256 bits) and AVX512(512 bits). 
Could you please see the [full explanation here](https://github.com/opencv/opencv/pull/21119#discussion_r756862412)?

Reply:
> > by why reorderning is here? please add comment
> 
> The difference links with lengths of SIMD vectors of SSE/NEON(128 bits), AVX2(256 bits) and AVX512(512 bits). Could you please see the [full explanation here](https://github.com/opencv/opencv/pull/21119#discussion_r756862412)?

For example let's compare SSE42 and AVX2. 
The function above handles 3-channel case. In current snippet I load float coefficients from `scalar` array.

**SSE42/NEON case**:
A vector has 128 bits. So it can contain 4 float (4*32 bits=128) numbers. Loading the first, the second and third vectors looks like these:

First: C1 C2 C3 **_C1_**
Second: **_C2_** C3 C1 **_C2_**
Third: **_C3_** C1 C2 C3

As you remember I load vectors from `scalar[]` array that look like this 
`float scalar[ nlanes+2 ] = {C1, C2, C3, C1, C2, C3 ...};`

Therefore, in for SSE/NEON/AVX512 the second SIMD vector should start loading since the second `scalar[] ` array element (`scalar[1] = C2`) and third SIMD vector should start loading at third `scalar[]` array element (`scalar[2] = C3`). Exception is AVX2.

**AVX2 case:**
A vector has 256 bits. So I can load to it 8 32-bit floats. So 3 vectors look like these:

First: C1 C2 C3 C1 C2 C3 C1 **_C2_**
Second: **_C3_** C1 C2 C3 C1 C2 C3 **_C1_**
Third: **_C2_** C3 C1 C2 C3 C1 C2 C3

So, in AVX2 case the second vector starts loading since third `scalar[]` element (`scalar[2] = C3`) and third vector starts loading at second `scalar[]` element (`scalar[1] = C2`).

So, It is the reason for the difference in this snippet.

Reply:
could you please put something short ( one line explanation) in code comment? but not in review, because no one checked out review comments when read a specific piece of code in IDE or notepad :)